lib/descriptors/MessageDescriptor.js

```diff
@@ -136,6 +140,37 @@ MessageDescriptor.prototype.getFieldNames = function () {
 }
 
 
+MessageDescriptor.prototype.getNextOneofIndex = function () {
+  return this._oneofs.length
+}
+
+
+MessageDescriptor.prototype.addOneof = function (oneof) {
```

Comment:
maybe the OneofDescriptor should be created here and returned, rather than passing it on and checking to make sure it's well-formed.